app/javascript/components/BalanceGraph.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import { LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer } from 'recharts';
+
+import { generateUniqueColors } from '../utils/generateUniqueColors';
+
+/* eslint react/prop-types: 0 */
+
+function formatDate(date) {
+  return new Date(date).toLocaleDateString();
+}
+
+const TooltipContent = ({ active, payload }) => {
+  if (active && payload && payload.length) {
+    const data = payload[0].payload;
+
+    return (
+      <div
+        className="custom-tooltip"
+        style={{ backgroundColor: 'rgba(0, 0, 0, 0.5)', padding: 10, textAlign: 'center', borderRadius: 8 }}
```

Comment:
The tooltip text can be kind of hard to read here. I'm guessing this was a preexisting convention but I played around with the tooltip background color a bit and I'm wondering that you think of this:

Original tooltip:
![tooltip_original_bg](https://user-images.githubusercontent.com/35817716/112764086-f068fa00-8fbb-11eb-9e2d-f5f4992d8379.png)

Proposed tooltip lighter background (`rgba(210, 210, 210, 0.5)`) with date as black text color (`#000000`):
![tooltip_proposed_bg_210_210_210_0 5](https://user-images.githubusercontent.com/35817716/112764130-14c4d680-8fbc-11eb-926b-5de9380983c1.png)

Reply:
There's a cool, new-ish CSS property called `backdrop-filter` that we may be able to leverage in the future for adding background blur to these tooltips without needing a parent element to house the blur property, and that could help the text really pop here. But at the moment, it's lacking major browser support.
https://developer.mozilla.org/en-US/docs/Web/CSS/backdrop-filter

Reply:
Agreed this looks better! I'll throw in the change.